control_panel_api/views.py

```diff
@@ -227,6 +233,25 @@ def perform_create(self, serializer):
 
         instance.create_users3bucket(user=self.request.user)
 
+    @detail_route()
+    def access_logs(self, request, pk=None):
+        query_params_serializer = S3BucketAccessLogsQueryParamsSerializer(
+            data=request.query_params)
+        query_params_serializer.is_valid(raise_exception=True)
+
+        es = ElasticSearch(settings.ELASTICSEARCH_CONN)
+
+        try:
+            result = es.bucket_hits_aggregation(
+                settings.ELASTICSEARCH_INDEX_S3LOGS,
+                self.get_object().name,
+                query_params_serializer.validated_data.get('day_range'),
+            )
+        except TransportError as e:
+            raise ESException(e) from e
```

Comment:
I don't know in which cases `TransportError` is raised and if it contains any useful details but it may be a good idea to log the original error message before swallowing/wrapping it into `ESException`.

Depending on how much details there are in the original error, having them in the logs may help in the future.

Reply:
The original is logged by the es library. The error message will get passed to this exception and stacktrace is maintaned due to the use of `from e`. This error is just a nice wrapper and enables the framework to handle error output.

Reply:
👍